components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/uaepass/UAEPassAuthenticator.java

```diff
@@ -646,17 +653,47 @@ protected String getLoginType(HttpServletRequest request) {
         return null;
     }
 
+    /**
+     * Replaces dynamic parameters in the query string with their corresponding values from the provided parameters map.
+     *
+     * @param queryString The query string containing dynamic parameters in the format ${parameterName}.
+     * @param parameters  A map of parameter names to their values.
+     * @return The query string with dynamic parameters replaced by their corresponding values.
+     */
+    private String replaceDynamicParams(String queryString, Map<String,String[]> parameters) {
+
+        Pattern pattern = Pattern.compile(DYNAMIC_PARAMETER_LOOKUP_REGEX);
```

Comment:
can we move this to class level? having it in the method level might have some perf impact

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-uaepass/pull/19/commits/16e1a3eb4ecf803d85db5aceb712b7e31190f9f0